File: 02-backend/spring-boot-ecommerce/spring-boot-ecommerce/src/main/java/com/tanmay/ecommerce/entity/OrderItem.java
package com.tanmay.ecommerce.entity;

import java.math.BigDecimal;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name="order_item")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class OrderItem {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id")
	private long id;
	
	@Column(name = "image_url")
	private String imageUrl;
	
	@Column(name = "unit_price")
	private BigDecimal unitPrice;
	
	@Column(name = "quantity")
	private int quantity;
	
	@Column(name = "product_id")
	private Long productId;
	
	@ManyToOne
	@JoinColumn(name = "order_id")
	private Order order;

}

File: 03-frontend/angular-ecommerce/src/app/common/order-item.ts
import { CartItem } from './cart-item';

export class OrderItem {

    imageUrl: string;
    unitPrice: number;
    quantity: number;
    productId: string;

    constructor(cardItem: CartItem) {
        this.imageUrl = cardItem.imageUrl;
        this.quantity = cardItem.quantity;
        this.unitPrice = cardItem.unitPrice;
        this.productId = cardItem.id;
    }
}
